.github/workflows/eventing-composition.yml

```diff
@@ -0,0 +1,75 @@
+name: Eventing composition tests
+
+on: 
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  
+env:
+  GOOS: linux
+  GO111MODULE: on
+
+jobs:
+  knative:
+    name: Eventing uBench (knative)
+    env:
+        GITHUB_RUN_ID: ${{ github.run_id }}
+        LOGPATH: /tmp/uBench-logs/${{ github.run_id }}
+        GOCACHE: /root/tmp/gocache
+        GOPATH: /root/tmp/gopath
+    runs-on: [self-hosted, cri]
+    strategy:
+      fail-fast: false
+    steps:
+
+    - name: Set up Golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.15
+
+    - name: Setup LOGPATH
+      run: |
+        mkdir -p /tmp/uBench-logs/${{ github.run_id }}
+
+    - name: Host Info
+      run: |
+        echo $HOSTNAME
+        echo $GITHUB_RUN_ID
+        echo $LOGPATH
+
+    - uses: actions/checkout@v2
+
+    - name: Start containerd
+      run: sudo containerd 1>$LOGPATH/uBench.out 2>$LOGPATH/uBench.err &
+
+    - name: Start one-node cluster
+      run : |
+        ./scripts/cluster/create_one_node_cluster.sh stock-only && sleep 2m
+
+    - name: Start consumer and producer
+      run: KUBECONFIG=/etc/kubernetes/admin.conf ./function-images/tests/eventing/sync-config/apply.sh && sleep 30s
+
+    - name: Install grpcurl
+      run: wget -qO- https://github.com/fullstorydev/grpcurl/releases/download/v1.8.1/grpcurl_1.8.1_linux_x86_64.tar.gz | sudo tar -C /usr/bin/ -xz grpcurl
```

Comment:
i think if we decide to use grpcurl, we should download its binary and place under `bins`, adding it to GitLFS (check .git* file in bins/). These release links are not reliable